reframe/utility/json.py

```diff
@@ -34,8 +34,8 @@ def __init__(self, *args, **kwargs):
                                   *args, **kwargs)
 
     def object_hook(self, obj):
-        if 'rfm_properties' in obj:
-            self.rfm_obj.__rfm_json_restore__(obj['rfm_properties'])
+        if 'modules' in obj:
```

Comment:
You should look at the "type" attribute instead and you should check that the `__rfm_json_restore__` is defined in the target object.